"tests" compile failure "cannot find type `KvTestEngine` in this scope"

## Development Task

#16506 moves test crates to `[dev-dependencies ]`, and it unexpected breaks feature propagation for cargo test commands, including features `test-engine-kv-rocksdb`, `test-engine-raft-raft-engine` etc.

```
➜ cargo test -p tests --features="failpoints test-engine-kv-rocksdb test-engine-raft-raft-engine" -- --nocapture --test test_txn_store_gc3
    Blocking waiting for file lock on build directory
   Compiling raftstore-v2 v0.1.0 (/home/stn/tikv/components/raftstore-v2)
   Compiling tikv_kv v0.1.0 (/home/stn/tikv/components/tikv_kv)
   Compiling engine_rocks_helper v0.1.0 (/home/stn/tikv/components/engine_rocks_helper)
   Compiling engine_test v0.0.1 (/home/stn/tikv/components/engine_test)
error[E0412]: cannot find type `KvTestEngine` in this scope
  --> components/engine_test/src/lib.rs:98:59
   |
98 |     pub fn new_engine(path: &str, cfs: &[&str]) -> Result<KvTestEngine> {
   |                                                           ^^^^^^^^^^^^ not found in this scope
   |
help: you might be missing a type parameter
   |
98 |     pub fn new_engine<KvTestEngine>(path: &str, cfs: &[&str]) -> Result<KvTestEngine> {
   |                      ++++++++++++++

error[E0412]: cannot find type `KvTestEngine` in this scope
   --> components/engine_test/src/lib.rs:106:17
    |
106 |     ) -> Result<KvTestEngine> {
    |                 ^^^^^^^^^^^^ not found in this scope
    |
help: you might be missing a type parameter
    |
102 |     pub fn new_engine_opt<KvTestEngine>(
    |                          ++++++++++++++

error[E0412]: cannot find type `KvTestEngine` in this scope
   --> components/engine_test/src/lib.rs:122:24
    |
122 |     impl TabletFactory<KvTestEngine> for TestTabletFactory {
    |                        ^^^^^^^^^^^^ not found in this scope
    |
help: you might be missing a type parameter
    |
122 |     impl<KvTestEngine> TabletFactory<KvTestEngine> for TestTabletFactory {
    |         ++++++++++++++

error[E0412]: cannot find type `KvTestEngine` in this scope
   --> components/engine_test/src/lib.rs:123:74
    |
123 |         fn open_tablet(&self, ctx: TabletContext, path: &Path) -> Result<KvTestEngine> {
    |                                                                          ^^^^^^^^^^^^ not found in this scope
    |
help: you might be missing a type parameter
    |
122 |     impl<KvTestEngine> TabletFactory<KvTestEngine> for TestTabletFactory {
    |         ++++++++++++++

error[E0412]: cannot find type `KvTestEngine` in module `crate::kv`
   --> components/engine_test/src/lib.rs:520:40
    |
520 | ) -> engine_traits::Engines<crate::kv::KvTestEngine, crate::raft::RaftTestEngine> {
    |                                        ^^^^^^^^^^^^ not found in `crate::kv`
    |
note: found an item that was configured out
   --> components/engine_test/src/lib.rs:86:24
    |
86  |         PanicEngine as KvTestEngine, PanicEngineIterator as KvTestEngineIterator,
    |                        ^^^^^^^^^^^^
    = note: the item is gated behind the `test-engine-kv-panic` feature
note: found an item that was configured out
   --> components/engine_test/src/lib.rs:91:24
    |
91  |         RocksEngine as KvTestEngine, RocksEngineIterator as KvTestEngineIterator,
    |                        ^^^^^^^^^^^^
    = note: the item is gated behind the `test-engine-kv-rocksdb` feature

warning: unused import: `MiscExt`
  --> components/engine_test/src/lib.rs:94:25
   |
94 |     use engine_traits::{MiscExt, Result, TabletContext, TabletFactory};
   |                         ^^^^^^^
   |
   = note: `#[warn(unused_imports)]` on by default

warning: unused import: `KvEngineConstructorExt`
  --> components/engine_test/src/lib.rs:96:64
   |
96 |     use crate::ctor::{CfOptions as KvTestCfOptions, DbOptions, KvEngineConstructorExt};
   |                                                                ^^^^^^^^^^^^^^^^^^^^^^

error[E0433]: failed to resolve: use of undeclared type `KvTestEngine`
  --> components/engine_test/src/lib.rs:99:9
   |
99 |         KvTestEngine::new_kv_engine(path, cfs)
   |         ^^^^^^^^^^^^ use of undeclared type `KvTestEngine`

error[E0433]: failed to resolve: use of undeclared type `KvTestEngine`
   --> components/engine_test/src/lib.rs:107:9
    |
107 |         KvTestEngine::new_kv_engine_opt(path, db_opt, cfs_opts)
    |         ^^^^^^^^^^^^ use of undeclared type `KvTestEngine`

error[E0433]: failed to resolve: use of undeclared type `KvTestEngine`
   --> components/engine_test/src/lib.rs:124:13
    |
124 |             KvTestEngine::new_tablet(
    |             ^^^^^^^^^^^^ use of undeclared type `KvTestEngine`

error[E0433]: failed to resolve: use of undeclared type `KvTestEngine`
   --> components/engine_test/src/lib.rs:138:13
    |
138 |             KvTestEngine::exists(path.to_str().unwrap_or_default())
    |             ^^^^^^^^^^^^ use of undeclared type `KvTestEngine`

Some errors have detailed explanations: E0412, E0433.
For more information about an error, try `rustc --explain E0412`.
warning: `engine_test` (lib) generated 2 warnings
error: could not compile `engine_test` (lib) due to 9 previous errors; 2 warnings emitted
warning: build failed, waiting for other jobs to finish...
```

--- a/components/engine_test/src/lib.rs
+++ b/components/engine_test/src/lib.rs
@@ -64,7 +64,13 @@ pub mod raft {
     #[cfg(feature = "test-engine-raft-rocksdb")]
     pub use engine_rocks::RocksEngine as RaftTestEngine;
     use engine_traits::Result;
-    #[cfg(feature = "test-engine-raft-raft-engine")]
+    #[cfg(any(
+        feature = "test-engine-raft-raft-engine",
+        not(any(
+            feature = "test-engine-raft-panic",
+            feature = "test-engine-raft-rocksdb"
+        ))
+    ))]
     pub use raft_log_engine::RaftLogEngine as RaftTestEngine;
 
     use crate::ctor::{RaftDbOptions, RaftEngineConstructorExt};
@@ -83,7 +89,10 @@ pub mod kv {
         PanicEngine as KvTestEngine, PanicEngineIterator as KvTestEngineIterator,
         PanicSnapshot as KvTestSnapshot, PanicWriteBatch as KvTestWriteBatch,
     };
-    #[cfg(feature = "test-engine-kv-rocksdb")]
+    #[cfg(any(
+        feature = "test-engine-kv-rocksdb",
+        not(feature = "test-engine-kv-panic")
+    ))]
     pub use engine_rocks::{
         RocksEngine as KvTestEngine, RocksEngineIterator as KvTestEngineIterator,
         RocksSnapshot as KvTestSnapshot, RocksWriteBatchVec as KvTestWriteBatch,

